### PR TITLE
feat!: skip verifying key derive in tests

### PR DESCRIPTION
**Pull request by emostov**
Changes
- breaking: add program ID to submit program request. This allows us to pass this back to the user when we want to skip program ID derives
- add option to opt out of checking program IDs. This isn't available in the CLI and we instead just configure it when setting up testing
- regen protos. Looks like we are on a new proto dep version so generated code is a little different

### CODE REVIEW

File: crates/coprocessor-node/src/event.rs [hunk 1]
```diff
@@ -163,11 +163,6 @@ where
                             let backoff = i * SUBMIT_JOB_BACKOFF_BASE_MS;
                             sleep(Duration::from_millis(backoff as u64)).await;
                         }
-
-                        error!(
```
Comment (emostov) [inline crates/coprocessor-node/src/event.rs:167]: this error was incorrect - it prints any time we successfully send a job requests

File: crates/scripts/src/local.rs [hunk 2]
```diff
@@ -132,21 +133,26 @@ async fn main() {
         let mut coproc_client =
             CoprocessorNodeClient::connect(client_coproc_grpc.clone()).await.unwrap();
         info!("Submitting CLOB ELF to coprocessor node");
-        let submit_program_request =
-            SubmitProgramRequest { program_elf: CLOB_ELF.to_vec(), vm_type: VmType::Sp1.into() };
+        let submit_program_request = SubmitProgramRequest {
+            program_elf: CLOB_ELF.to_vec(),
+            vm_type: VmType::Sp1.into(),
+            program_id: get_matching_game_program_id().to_vec(),
```
Comment (MaanavKhaitan) [inline crates/scripts/src/local.rs:139]: this should be `get_clob_program_id()`

File: crates/sdk/proto/src/descriptor.bin
Comment (MaanavKhaitan) [inline crates/sdk/proto/src/descriptor.bin:1]: should this be committed to git?

**Review by MaanavKhaitan — APPROVED** [review]
LGTM, a few nits

File: crates/sdk/proto/src/descriptor.bin
Comment (emostov) [reply crates/sdk/proto/src/descriptor.bin:1]: ya we have been doing it so far

File: crates/scripts/src/local.rs (hunk 2 above)
Comment (emostov) [reply crates/scripts/src/local.rs:139]: nice ty

**Pull request merged**